agents/azure_arm/fence_azure_arm.py

```diff
@@ -115,10 +115,22 @@ def set_power_status(clients, options):
 
         if (options["--action"]=="off"):
             logging.info("Poweroff " + vmName + " in resource group " + rgName)
-            compute_client.virtual_machines.power_off(rgName, vmName, skip_shutdown=True)
+            try:
+                # try new API version first
+                compute_client.virtual_machines.begin_power_off(rgName, vmName, skip_shutdown=True)
+            except AttributeError:
+                # use older API verson if it fails
+                logging.info("Poweroff " + vmName + " did not work via 'virtual_machines.begin_power_off. Trying virtual_machines.power_off'.")
+                compute_client.virtual_machines.power_off(rgName, vmName, skip_shutdown=True)
         elif (options["--action"]=="on"):
             logging.info("Starting " + vmName + " in resource group " + rgName)
-            compute_client.virtual_machines.start(rgName, vmName)
+            try:
+                # try new API version first
+                compute_client.virtual_machines.begin_start(rgName, vmName)
+            except AttributeError:
+                # use older API verson if it fails
+                logging.info("Starting " + vmName + " did not work via 'virtual_machines.begin_start. Trying virtual_machines.start'.")
```

Comment:
These should be debug level, so it doesnt spam the logs unless you set HA_debug=1 or similar.

Reply:
fixed in a0f5193c323d04beb64defca4d6b4bca3631c0b5